pkg/asset/ignition/machine/masterignitioncheck.go

```diff
@@ -0,0 +1,94 @@
+package machine
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/tls"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+const (
+	masterMachineConfigFileName = "manifests/99_openshift-installer-ignition_master.yaml"
+)
+
+// MasterIgnitionCheck is an asset that checks for any customizations a user might
+// have made to the pointer ignition configs before creating the cluster. If customizations
+// are made, then the updates are reconciled as a MachineConfig file
+type MasterIgnitionCheck struct {
+	File *asset.File
+}
+
+var _ asset.WritableAsset = (*MasterIgnitionCheck)(nil)
+
+// Dependencies returns the dependencies for MasterIgnitionCheck
+func (a *MasterIgnitionCheck) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+		&tls.RootCA{},
+		&Master{},
+	}
+}
+
+// Generate queries for input from the user.
+func (a *MasterIgnitionCheck) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	rootCA := &tls.RootCA{}
+	master := &Master{}
+	dependencies.Get(installConfig, rootCA, master)
+
+	defaultPointerIgnition := pointerIgnitionConfig(installConfig.Config, rootCA.Cert(), "master")
+	savedPointerIgnition := master.Config
+
+	if savedPointerIgnition != defaultPointerIgnition {
+		logrus.Infof("Master pointer ignition was modified. Saving contents to a machineconfig")
+		mc := &mcfgv1.MachineConfig{}
+		mc, err := generatePointerMachineConfig(*savedPointerIgnition, "master")
+		if err != nil {
+			return errors.Wrap(err, "failed to generate master installer machineconfig")
+		}
+		configData, err := yaml.Marshal(mc)
+		if err != nil {
+			return errors.Wrap(err, "failed to marshal master installer machineconfig")
+		}
+		a.File = &asset.File{
+			Filename: masterMachineConfigFileName,
+			Data:     configData,
+		}
+	}
+
+	return nil
+}
+
+// Name returns the human-friendly name of the asset.
+func (a *MasterIgnitionCheck) Name() string {
+	return "Master Ignition Customization Check"
+}
+
+// Files returns the files generated by the asset.
+func (a *MasterIgnitionCheck) Files() []*asset.File {
+	if a.File != nil {
+		return []*asset.File{a.File}
+	}
+	return []*asset.File{}
+}
+
+// Load returns the master ignitions from disk.
+func (a *MasterIgnitionCheck) Load(f asset.FileFetcher) (found bool, err error) {
+	file, err := f.FetchByName(masterMachineConfigFileName)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
```

Comment:
Given this consumes the pointer config, I'm not sure we ever expect this to be loaded from file, so perhaps we should make that explicit and just always `return false, nil` here?